pkg/workload/workload_test.go

```diff
@@ -647,3 +649,71 @@ func TestResourceUsage(t *testing.T) {
 		})
 	}
 }
+
+func TestAdmissionCheckStrategy(t *testing.T) {
+	cases := map[string]struct {
+		cq                  *kueue.ClusterQueue
+		wl                  *kueue.Workload
+		wantAdmissionChecks sets.Set[string]
+	}{
+		"AdmissionCheckStrategy with a flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(*utiltesting.MakeAdmissionCheckStrategyRule("ac1", "flavor1").Obj()).
+				Obj(),
+			wantAdmissionChecks: sets.New("ac1"),
+		},
+		"AdmissionCheckStrategy with an unmatched flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(*utiltesting.MakeAdmissionCheckStrategyRule("ac1", "unmatched-flavor").Obj()).
+				Obj(),
+			wantAdmissionChecks: nil,
+		},
+		"AdmissionCheckStrategy without a flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(*utiltesting.MakeAdmissionCheckStrategyRule("ac1").Obj()).
+				Obj(),
+			wantAdmissionChecks: sets.New("ac1"),
+		},
+		"Two AdmissionCheckStrategies, one with flavor, one without flavor": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				ReserveQuota(utiltesting.MakeAdmission("cq").Assignment("cpu", "flavor1", "1").Obj()).
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(
+					*utiltesting.MakeAdmissionCheckStrategyRule("ac1", "flavor1").Obj(),
+					*utiltesting.MakeAdmissionCheckStrategyRule("ac2").Obj()).
+				Obj(),
+			wantAdmissionChecks: sets.New("ac1", "ac2"),
+		},
+		"Workload has no QuotaReserved": {
+			wl: utiltesting.MakeWorkload("wl", "ns").
+				Obj(),
+			cq: utiltesting.MakeClusterQueue("cq").
+				AdmissionCheckStrategy(
+					*utiltesting.MakeAdmissionCheckStrategyRule("ac1", "flavor1").Obj(),
+					*utiltesting.MakeAdmissionCheckStrategyRule("ac2").Obj()).
+				Obj(),
+			wantAdmissionChecks: nil,
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			_, log := utiltesting.ContextWithLog(t)
+			gotAdmissionChecks := AdmissionChecksForWorkload(log, tc.wl, utilac.NewAdmissionChecks(tc.cq))
+
+			if diff := cmp.Diff(tc.wantAdmissionChecks, gotAdmissionChecks); diff != "" {
```

Comment:
```suggestion
			if diff := cmp.Diff(tc.wantAdmissionChecks, gotAdmissionChecks, cmpopts.SortMap(...)); diff != "" {
```

This could have a potential flaky test since the `map` doesn't guarantee the order.

Reply:
but both are sets

Reply:
I don't think this is the case, since we compare `sets.Set` and it implements `Equal` method which is invoked during comparison

Reply:
IIUC, the sets are typed `map[T]struct{}`. So, could we re-order by the `cmpopts.SortMap`.
@PBundyra, could you investigate them?

Reply:
> I don't think this is the case, since we compare sets.Set and it implements Equal method which is invoked during comparison

Oh, I didn't know that. Thank you for the clarification!